barretenberg/cpp/src/barretenberg/protogalaxy/presumcheck_prover.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <utility>
+
+#include "barretenberg/flavor/goblin_ultra.hpp"
+#include "barretenberg/flavor/ultra.hpp"
+#include "barretenberg/sumcheck/instance/prover_instance.hpp"
+#include "barretenberg/transcript/transcript.hpp"
+
+namespace bb {
+
+template <IsUltraFlavor Flavor> class PreSumcheckProver {
```

Comment:
Can you please add documentation to this class.. also maybe we should discuss whether a prover inside a prover is the right approach, I was envisioning an architecture where we have some shared rounds that can be in a utility class, similar to the utility class for operating on relations that both sumcheck and protogalaxy uses. It would be nice to have this class be static but might require some more refactoring work.

Reply:
will add more comments. Prover inside prover is already our current standard. We have a SumcheckProver and ZeromorphProver both inside UltraProver.

I agree that the PreSumCheckProver can be split into further classes for each Round, and can eventually be made static or something like that.

Reply:
That is a good point, I forgot about those. What do you think about creating a PreSumchVerifier as well (since we have a SumcheckVerifier and ZeromorphVerifier)? If you prefer not to do this in this PR, I think an issue should be added

Reply:
I think that can just go into a followup PR.

Reply:
ok, please add an issue